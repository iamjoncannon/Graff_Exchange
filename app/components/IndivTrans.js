import React from 'react';

const dummyData = [
  ["Buy", "4", "1152", "2019-07-16"],
  ["Buy", "4", "1152", "2019-07-16"],
  ["Buy", "4", "1152", "2019-07-16"],
  ["Buy", "4", "1152", "2019-07-16"],
  ["Buy", "4", "1152", "2019-07-16"],
  ["Buy", "4", "1152", "2019-07-16"],
  ["Buy", "4", "1152", "2019-07-16"],
  ["Buy", "4", "1152", "2019-07-16"],
  ["Buy", "4", "1152", "2019-07-16"],
  ["Buy", "4", "1152", "2019-07-16"],
  ["Buy", "4", "1152", "2019-07-16"],
  ["Buy", "4", "1152", "2019-07-16"],
  ["Buy", "4", "1152", "2019-07-16"],
  ["Buy", "4", "1152", "2019-07-16"],
  ["Buy", "4", "1152", "2019-07-16"],
  ["Buy", "4", "1152", "2019-07-16"],
  ["Buy", "4", "1152", "2019-07-16"],
  ["Buy", "4", "1152", "2019-07-16"],
  ["Buy", "4", "1152", "2019-07-16"],
  ["Buy", "4", "1152", "2019-07-16"],
  ["Buy", "4", "1152", "2019-07-16"],
  ["Buy", "4", "1152", "2019-07-16"],
  ["Buy", "4", "1152", "2019-07-16"],
  ["Buy", "4", "1152", "2019-07-16"],
  ["Buy", "4", "1152", "2019-07-16"],
  ["Buy", "4", "1152", "2019-07-16"],
  ["Buy", "4", "1152", "2019-07-16"],
  ["Buy", "4", "1152", "2019-07-16"],
  ["Buy", "4", "1152", "2019-07-16"],
  ["Buy", "4", "1152", "2019-07-16"],
  ["Buy", "4", "1152", "2019-07-16"],
  ["Buy", "4", "1152", "2019-07-16"],
  ["Buy", "4", "1152", "2019-07-16"],
  ["Buy", "4", "1152", "2019-07-16"],
  ["Buy", "4", "1152", "2019-07-16"],
  ["Buy", "4", "1152", "2019-07-16"],
  ["Buy", "4", "1152", "2019-07-16"],
  ["Buy", "4", "1152", "2019-07-16"],
  ["Buy", "4", "1152", "2019-07-16"],
  ["Buy", "4", "1152", "2019-07-16"],
  ["Buy", "4", "1152", "2019-07-16"],
  ["Buy", "4", "1152", "2019-07-16"],
  ["Buy", "4", "1152", "2019-07-16"],
  ["Buy", "4", "1152", "2019-07-16"],
  ["Buy", "4", "1152", "2019-07-16"],
  ["Buy", "4", "1152", "2019-07-16"],
  ["Buy", "4", "1152", "2019-07-16"],
  ["Buy", "4", "1152", "2019-07-16"],
  ["Buy", "4", "1152", "2019-07-16"],
  ["Buy", "4", "1152", "2019-07-16"],
]

export default class IndivTrans extends React.Component {
 
  constructor(props) {
    super(props);
    this.state = {

    }
  }

  componentDidMount(){

  }

  render() {

    return (
      <div className="indiv-trans indiv-container">

        <div>

          <span>Type</span>
          <span>Quantity</span>
          <span>Price</span>
          <span>Date</span>

        </div>

        <div>

          {
            dummyData.map((item, i)=>{
              return(

                <div>
                  <span>{item[0]}</span>
                  <span>{item[1]} </span>
                  <span>{item[2]}</span>
                  <span>{item[3]}</span>
                </div>
              )
            })
          }
        </div>
      </div>
    );
  }
}

