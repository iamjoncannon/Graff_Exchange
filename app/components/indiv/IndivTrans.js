import React from 'react';
import { connect } from "react-redux";

class IndivTrans extends React.Component {
 
  constructor(props) {
    super(props);
    this.state = {

    }
  }

  componentDidMount(){

  }

  render() {

    const {transactionHistory, selectedPortfolioItem} = this.props

    return (
      <div className="indiv-trans indiv-container">

        <div>

          <span>Type</span>
          <span>Quantity</span>
          <span>Price</span>
          <span>Date</span>

        </div>

        <div>

          { transactionHistory &&
            Object.entries(transactionHistory)
              .filter(item =>{                
                console.log(item)
                return item[1].Symbol === selectedPortfolioItem.symbol && item[1].Quantity > 0
              })
              .map((item, i)=>{

                let date = item[1].Date 
                date = date.slice(0,date.indexOf("T"))

                return(

                  <div>
                    <span>{item[1].Type}</span>
                    <span>{item[1].Quantity}</span>
                    <span>${item[1].Price}</span>
                    <span>{date}</span>
                  </div>
                )
            })
          }
        </div>
      </div>
    );
  }
}


const mapStateToProps = ({ Portfolio_state }) => {

  return {
    transactionHistory : Portfolio_state.transactionHistory,
    selectedPortfolioItem: Portfolio_state.selectedPortfolioItem
  };
};

export default connect(
  mapStateToProps,
  null
)(IndivTrans);
