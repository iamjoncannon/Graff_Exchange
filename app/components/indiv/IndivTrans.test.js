import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import IndivTrans from './IndivTrans'
import configureStore from 'redux-mock-store';
import thunk from "redux-thunk"
const mockStore = configureStore([thunk]);
import { BrowserRouter as Router } from "react-router-dom";

const fake_store = {
    Portfolio_state: {
        transactionHistory : [{
            date_conducted: "Wed Jul 17 2019 20:13:21 GMT-0400 (Eastern Daylight Time)",
            id: "14",
            price: 10.329999923706055,
            quantity: 100,
            symbol: "F",
            type: "Buy"
        }],
        selectedPortfolioItem: "F"
    }
}

describe("IndivTrans", ()=>{

    let component 
    let store 
    let connected_react_component

    beforeEach(()=>{
                
        store = mockStore( fake_store );
    
        component = renderer.create(
            <Provider store={store} >
               <Router >
                   <IndivTrans pathname={"/"} />
               </Router>
           </Provider>)

        connected_react_component = component.root.children[0].children[0].children[0].children[0]._fiber.stateNode

    })

    it('1) renders successfully with data from redux store', ()=>{

        let tree = component.toJSON();
        
        expect(tree).toMatchSnapshot();
    })
    
    it("2) renders four title spans", ()=>{

        const className = "indiv-trans indiv-container"

        let parent = component.root.find( node=> node.props.className === className)

        let spans = parent.children[0].props.children.map(child => child.props.children)
        
        expect(spans).toEqual( ['Type', 'Quantity', 'Price', 'Date' ])
    })

    it("3) renders transaction history item with four spans", ()=>{

        const className = "indiv-trans indiv-container"

        let parent = component.root.find( node=> node.props.className === className)

        let spans = parent.children[1].props.children[0].props.children.map(child => child.props.children)

        expect(spans).toEqual([ 'Buy', 100, [ '$', 10.329999923706055 ], 'Wed Jul 17 2019 ' ])
  
    })
})
