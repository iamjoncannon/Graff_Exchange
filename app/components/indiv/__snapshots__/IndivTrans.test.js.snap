// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndivTrans 1) renders successfully with data from redux store 1`] = `
<div
  className="indiv-trans indiv-container"
>
  <div>
    <span>
      Type
    </span>
    <span>
      Quantity
    </span>
    <span>
      Price
    </span>
    <span>
      Date
    </span>
  </div>
  <div>
    <div>
      <span>
        Buy
      </span>
      <span>
        100
      </span>
      <span>
        $
        10.329999923706055
      </span>
      <span>
        Wed Jul 17 2019 
      </span>
    </div>
  </div>
</div>
`;
